### PR TITLE
[MRG] change the threshold for the rank in RAP-MUSIC

### PR DESCRIPTION
**Pull request by yousrabk**
When using another set of phantom data, the threshold used to estimate the rank is not good. It makes the rank=3 which is not possible in sphere models causing wrong estimation of dipole orientation.

### CODE REVIEW

File: mne/beamformer/_rap_music.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def _compute_subcorr(G, phi_sig):
     # in G and handle the fact that it might be rank defficient
     # eg. when using MEG and a sphere model for which the
     # radial component will be truly 0.
-    rank = np.sum(Sg > (Sg[0] * 1e-12))
+    rank = np.sum(Sg > (Sg[0] * 1e-6))
```
Comment (larsoner) [inline mne/beamformer/_rap_music.py]: At this point in the code, is `Sg` whitened or rescaled by channel type? If not, this could cause problems for M-EEG data, as the gain matrix could differ by a factor this large IIRC.

**Comment by yousrabk** [comment]
@larsoner @agramfort there was already a discussion between you two about this threshold (1e-6 or 1e-12). wdyt?
capture d ecran 2017-10-16 a 17 31 22

File: mne/beamformer/_rap_music.py (hunk 1 above)
Comment (yousrabk) [reply mne/beamformer/_rap_music.py]: Yes at this point, G (so Sg) is already whitened

**Comment by larsoner** [comment]
If the gain matrix was computed or saved/loaded using single precision `1e-6` is better, so we can just use it. Is that what happened?

**Comment by yousrabk** [comment]
How would you specify the precision when computing/saving a forward operator?

**Comment by larsoner** [comment]
We always compute in double precision, but save in single precision

**Comment by yousrabk** [comment]
Ok got it. For my data, I actually compute the forward in advance and save it. However for the bst_phantom as in the tutorial, I just compute the forward and I use it directly.

**Comment by yousrabk** [comment]
So I tried to save the forward before using it with the bst_phantom_elekta dataset and we have the same problem. Here are two plots to have an idea of what I mean by problematic in estimating the ori and the amplitude of the dipoles.
This first plot is showing the errors in (1) Location (2) Orientation (3) amplitude using master:
capture d ecran 2017-10-16 a 18 36 29
The errors in ori/amp are so big that they are out of range of an "acceptable" error.

This second one is showing the same errors using the threshold 1e-6 instead of 1e-12 (this branch):
capture d ecran 2017-10-16 a 18 13 03

5-6-7-8 are the index of the dipole. 20-100-200-1000nAm are the 4 different amplitude of the simulated dipole (4 different evokeds).

Basically, it is due to the single precision when reading back the forward.

**Comment by larsoner** [comment]
Yep, that's what I'd expect. So +1 for merge from me

**Comment by agramfort** [comment]
@yousrabk please update what's new with a bug fix entry and +1 for MRG

**Comment by agramfort** [comment]
thx @yousrabk
